src/openboardview/FileFormats/BVR3File.cpp

```diff
@@ -0,0 +1,207 @@
+
+	ENSURE_OR_FAIL(buffer_size > 4, error_msg, return);
+	size_t file_buf_size = 3 * (1 + buffer_size);
+	file_buf             = (char *)calloc(1, file_buf_size);
+	ENSURE_OR_FAIL(file_buf != nullptr, error_msg, return);
+
+	std::copy(buf.begin(), buf.end(), file_buf);
+	file_buf[buffer_size] = 0;
+	// This is for fixing degenerate utf8
+	char *arena     = &file_buf[buffer_size + 1];
+	char *arena_end = file_buf + file_buf_size - 1;
+	*arena_end      = 0;
+
+	BRDPart blank_part;
+	BRDPin blank_pin;
+	BRDPart part;
+	BRDPin pin;
+	std::list<std::pair<BRDPoint, BRDPoint>> outline_segments;
+
+	std::vector<char *> lines;
+	stringfile(file_buf, lines);
+
+	for (char *line : lines) {
+		while (isspace((uint8_t)*line)) line++;
+		if (!line[0]) continue;
+
+		char *p = line;
+		char *s;
+
+		if (!strncmp(line, "PART_NAME ", 10)) {
+			p += 10;
+			part.name = READ_STR();
+		} else if (!strncmp(line, "PART_SIDE ", 10)) {
+			p += 10;
+			char *side = READ_STR();
+			if (!strcmp(side, "T"))
+				part.mounting_side = BRDPartMountingSide::Top;
+			else
+				part.mounting_side = BRDPartMountingSide::Bottom;
+		} else if (!strncmp(line, "PART_ORIGIN ", 12)) {
+			p += 12;
+			double origin_x = READ_DOUBLE();
+			part.p1.x = trunc(origin_x);
+			double origin_y = READ_DOUBLE();
+			part.p1.y = trunc(origin_y);
+		} else if (!strncmp(line, "PART_MOUNT ", 11)) {
+			p += 11;
+			char *mount = READ_STR();
+			if (!strcmp(mount, "SMD"))
+				part.part_type = BRDPartType::SMD;
+			else
+				part.part_type = BRDPartType::ThroughHole;
+		} else if (!strncmp(line, "PIN_ID ", 7)) {
+			p += 7;
+			pin.snum = READ_STR();
+			pin.part = parts.size() + 1; // pin is for current part, which will not yet have been added to parts vector
+		} else if (!strncmp(line, "PIN_NUMBER ", 11)) {
+			// Ignored, not sure if relevant for BRDPin
+		} else if (!strncmp(line, "PIN_NAME ", 9)) {
+			p += 9;
+			pin.name = READ_STR();
+		} else if (!strncmp(line, "PIN_SIDE ", 9)) {
+			// Ignored, not sure if relevant for BRDPin
+		} else if (!strncmp(line, "PIN_ORIGIN ", 11)) {
+			p += 11;
+			double origin_x = READ_DOUBLE();
+			pin.pos.x = trunc(origin_x);
+			double origin_y = READ_DOUBLE();
+			pin.pos.y = trunc(origin_y);
+		} else if (!strncmp(line, "PIN_RADIUS ", 11)) {
+			p += 11;
+			pin.radius = READ_DOUBLE();
+		} else if (!strncmp(line, "PIN_NET ", 8)) {
+			p += 8;
+			pin.net = READ_STR();
+		} else if (!strncmp(line, "PIN_TYPE ", 9)) {
+			// Ignored, not sure if relevant for BRDPin
+		} else if (!strncmp(line, "PIN_COMMENT ", 12)) {
+			// Ignored, not sure if relevant for BRDPin
+		} else if (!strncmp(line, "PIN_OUTLINE_RELATIVE ", 21)) {
+			// Ignored, not sure if relevant for BRDPin
+		} else if (!strcmp(line, "PIN_END")) {
+			pins.push_back(pin);
+			pin = blank_pin;
+		} else if (!strcmp(line, "PART_END")) {
+			part.end_of_pins = pins.size();
+			parts.push_back(part);
+			part = blank_part;
+		} else if (!strncmp(line, "OUTLINE_SEGMENTED ", 18)) {
```

Comment:
Some files use `OUTLINE_POINTS` instead which is not handled here.

Reply:
Thanks, will check other files and implement.

Reply:
Implemented, looks good to me now on the files I've tried that use OUTLINE_POINTS but would appreciate you checking to confirm.